.github/workflows/publish-backend-plugin-manager.yaml

```diff
@@ -50,21 +50,33 @@ jobs:
           else
             GIT_REF=master
           fi
-          echo "GIT_REF=$GIT_REF" >> $GITHUB_ENV
+          echo "GIT_REF=$GIT_REF" >> $GITHUB_OUTPUT
 
-      - name: Checkout
+      - name: Checkout from master
+        if: steps.get_checkout_ref.outputs.GIT_REF == 'master'
         uses: actions/checkout@v4
         with:
           repository: backstage/backstage
-          ref: ${{ steps.get_checkout_ref.outputs.GIT_REF }}
+          ref: master
           # fetch all history, but only for the current ref
           fetch-depth: 2147483647
```

Comment:
Seems weird to need the full history. why not just the last 20 commits ?

Reply:
Because we want to be able to search for the last commit that impacted the package `backend-plugin-manager`. I just didn't want to use an arbitrary limit that might return a wrong commit.
OTOH we only checkout one branch, so that's not that long.